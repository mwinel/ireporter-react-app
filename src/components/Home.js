import React, { Component } from "react";
import Navbar from '../components/Layout/Navbar';
import Footer from '../components/Layout/Footer';

class Home extends Component {
  render () {
    return (
      <div>
        <Navbar />
        <div className="container">
          <div className="row">
            <div className="flag-content">
              <h1>Welcome to iReporter!</h1>
            </div>
          </div>
          <Footer />
        </div>
      </div>
    );
  }
}

export default Home;
