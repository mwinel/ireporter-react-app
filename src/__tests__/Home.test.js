import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import Home from '../components/Home';

const storeItems = {};
const middlewares = [thunk]
const mockStore = configureStore(middlewares);
const store = mockStore(storeItems);

const props = {
  fetchIncidents: jest.fn(),
  getState: jest.fn()
}

describe('Home Component', () => {
  let wrapper = shallow(<Home {...props} store={store} />)
  it('Should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
