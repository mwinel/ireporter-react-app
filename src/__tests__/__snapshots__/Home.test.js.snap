// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Should match snapshot 1`] = `
<ContextProvider
  value={null}
>
  <Home
    fetchIncidents={[Function]}
    getState={[MockFunction]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
